### PR TITLE
Switch some scripts to ESM

### PR DESCRIPTION
**Pull request by fisker**
## Description

Need update https://github.com/prettier/prettier/wiki/Release-Checklist after merge.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/build/utils.mjs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      return url.fileURLToPath(importMeta.url);
+    },
+    get __dirname() {
+      return path.dirname(url.fileURLToPath(importMeta.url));
+    },
+    get require() {
+      return createRequire(importMeta.url);
+    },
+  };
+}
```
Comment (sosukesuzuki) [inline scripts/build/utils.mjs:33]: This is super useful utils for me, so I published the package [`esm-migration-utils`](https://github.com/sosukesuzuki/esm-migration-utils). Probably we will use this several times in the future in our codes. Can we use the package?

Are you sure you want to list your name in the README? like `This package is extracted from fisker's code.`Also, I can remove this package if you don't like it.

Comment (fisker) [reply scripts/build/utils.mjs:33]: I have `esm-utils` module, haha.

Comment (sosukesuzuki) [reply scripts/build/utils.mjs:33]: Oh, I think we can use it on Prettier.

Comment (fisker) [reply scripts/build/utils.mjs:33]: > Are you sure you want to list your name in the README? like This package is extracted from fisker's code.

Not necessary

> Also, I can remove this package if you don't like it.

Feel free to do whatever with it.

Comment (fisker) [reply scripts/build/utils.mjs:33]: > I think we can use it on Prettier.

I don't like promoting my own packages, but I just published `v1.0.0`, I'll use it in this PR.

**Review by sosukesuzuki — APPROVED** [review]
Need more review

**Review by thorn0 — APPROVED** [review]

**Pull request merged**